src/daemon/http/server.rs

```diff
@@ -511,53 +512,59 @@ pub async fn metrics(req: Request) -> RoutingResult {
                 if let Ok(repo_status) = server.ca_repo_status(ca).await {
                     ca_repo_status.insert(ca.clone(), repo_status);
                 }
-            }
 
-            res.push('\n');
-            res.push_str("# HELP krill_cas_bgp_announcements_valid number of announcements seen for CA resources with RPKI state VALID\n");
-            res.push_str("# TYPE krill_cas_bgp_announcements_valid gauge\n");
-            for (ca, nr) in all_bgp_stats.announcements_valid.iter() {
-                res.push_str(&format!("krill_cas_bgp_announcements_valid{{ca=\"{}\"}} {}\n", ca, nr));
+                if let Ok(child_status_map) = server.ca_child_status_map(ca).await {
+                    ca_child_status.insert(ca.clone(), child_status_map);
+                }
             }
 
-            res.push('\n');
-            res.push_str("# HELP krill_ca_parent_success status of last ca-parent connection (0=issue, 1=success)\n");
-            res.push_str("# TYPE krill_ca_parent_success gauge\n");
-            for (ca, parent_statuses) in ca_parent_statuses.iter() {
-                if ca.as_str() != TA_NAME {
-                    for (parent, status) in parent_statuses.iter() {
-                        // skip the ones for which we have no status yet, i.e it was really only just added
-                        // and no attempt to connect has yet been made.
-                        if let Some(exchange) = status.last_exchange() {
-                            let value = if exchange.was_success() { 1 } else { 0 };
-                            res.push_str(&format!(
-                                "krill_ca_parent_success{{ca=\"{}\", parent=\"{}\"}} {}\n",
-                                ca, parent, value
-                            ));
+            {
```

Comment:
It's a bit hard to see from the GH diff but does each last success time metric have a corresponding last connection time metric?